### DISCUSSION TITLE
Deserialize to Expando?

### DISCUSSION THREAD

**Discussion by dkackman** [aaubry/YamlDotNet#617] [Q&A]
I have a large YAML document that I want to deserialize to a dynamic object (`ExpandoObject`). I can get there by going to Json first and then using NewtonSoft. Can YamlDotNet deserialize directly to a dynamic object tree?

    using var input = new StreamReader(fullPath);
    var deserializer = new DeserializerBuilder()
        .Build();
    var yamlObject = deserializer.Deserialize(input);
    var serializer = new SerializerBuilder()
        .JsonCompatible()
        .Build();

    var json = serializer.Serialize(yamlObject);
    dynamic config = JsonConvert.DeserializeObject<ExpandoObject>(json, new ExpandoObjectConverter());

**Comment by EdwardCooke** [comment]
Not that I am aware of. There is another project, it's pretty old and doesn't look like it supports anything other than full framework applications (no dot net core). You could have a look at this and see if it would work for you.

https://github.com/aaubry/YamlDotNet.Dynamic
